docs/installation/sources/sources_linux.rst

```diff
@@ -111,6 +110,34 @@ For example, on Ubuntu use the command:
 
    sudo apt install libssl-dev
 
+.. _libp11_sl:
+
+Libp11 and SoftHSM libraries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Libp11 provides PKCS#11 support for OpenSSL. This is an optional dependency,
+that is needed only when *eprosima Fast DDS* is used with security and PKCS#11 URIs.
+
+Install libp11_ using the package manager of the appropriate Linux distribution.
+For example, on Ubuntu use the command:
+
+.. code-block:: bash
+
+   sudo apt install libp11-dev libengine-pkcs11-openssl
+
+SoftHSM is a software implementation of an HSM (Hardware Security Module).
+If *eProsima Fast DDS* tests are activated and *libp11* is installed
+on the system, SoftHSM is additioanlly required to run tests of PKCS#11 features.
+
+To install SoftHSM_, download the repository and follow the installation instructions:
+
+.. code-block:: bash
+
+   git clone https://github.com/opendnssec/SoftHSMv2.git
+   cd SoftHSMv2 && ./autogen.sh
+   ./configure --with-crypto-backend=openssl --disable-non-paged-memory
+   make && sudo make install
```

Comment:
I have installed it directly from the apt repository when I run the tests locally (in fact the configuration you mention to make openSSL aware of the engine, I did not have to do it. It seems it is done directly in Ubuntu because I could run the tests without any trouble. The only issue was that I had to run the tests with root privileges because otherwise softHSM2 complains):

```
sudo apt install softhsm2
```